modules/javafx.base/src/main/java/javafx/collections/FXCollections.java

```diff
@@ -347,9 +347,9 @@ public static <E> ObservableList<E> observableArrayList(Callback<E, Observable[]
      * @see #observableArrayList()
      */
     public static <E> ObservableList<E> observableArrayList(E... items) {
-        ObservableList<E> list = observableArrayList();
-        list.addAll(items);
-        return list;
+        ArrayList<E> backingList = new ArrayList<>(items.length);
+        backingList.addAll(Arrays.asList(items));
```

Comment:
Unless I'm missing something again, this can be written as
`ArrayList<E> backingList = new ArrayList<>(Arrays.asList(items));`

I created benchmarks based on yours, and there is no significant difference between these two (their scores are within the error ranges of each other).

I don't mind leaving it as is if you think it's clearer, I personally prefer the 1 line variant.

Reply:
Thanks. I totally missed that, I prefer the 1 line variant as well.